src/expressions/operators/arithmetic/BinaryOperator.ts

```diff
@@ -122,6 +122,20 @@ export function generateBinaryOperatorType(
 	typeA: ValueType,
 	typeB: ValueType
 ): ValueType {
+	const GENERIC_NUMERIC_VALUE_TYPES = [
+		ValueType.XSNUMERIC,
+		ValueType.NODE,
+		ValueType.ITEM,
+		ValueType.XSANYATOMICTYPE,
+	];
+
+	if (
+		GENERIC_NUMERIC_VALUE_TYPES.includes(typeA) ||
+		GENERIC_NUMERIC_VALUE_TYPES.includes(typeB)
+	) {
+		return ValueType.XSNUMERIC;
+	}
```

Comment:
What is this snippet doing actually? If both left right types match one of the four we return numeric?

Reply:
If either left or right is one of the four ValueTypes, we return an XSNUMERIC